packages/graphql/src/schema/validation/custom-rules/utils/document-validation-error.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type {
+    DirectiveNode,
+    ObjectTypeDefinitionNode,
+    FieldDefinitionNode,
+    InterfaceTypeDefinitionNode,
+} from "graphql";
+
+export type AssertionResponse = {
+    isValid: boolean;
+    errorMsg?: string;
+    errorPath: ReadonlyArray<string | number>;
+};
+export type VALIDATION_FN = ({
```

Comment:
Feels strange to have this in upper case.

```suggestion
export type ValidationFunction = ({
```